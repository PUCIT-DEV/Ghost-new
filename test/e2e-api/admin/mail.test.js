const {agentProvider, fixtureManager, matchers, mockManager} = require('../../utils/e2e-framework');
const {anyEtag} = matchers;

describe('Mail API', function () {
    let agent;

    before(async function () {
        agent = await agentProvider.getAdminAPIAgent();
        await fixtureManager.init('invites');
        await agent.loginAsOwner();
    });

    beforeEach(function () {
        mockManager.mockMail({message: 'sent'});
    });

    afterEach(function () {
        mockManager.restore();
    });

    it('Can send mail', async function () {
        await agent
            .post('mail/')
            .body({
                mail: [{
                    message: {
                        to: 'joe@example.com',
                        subject: 'testemail',
                        html: '<p>This</p>'
                    }
                }]
            })
            .expectStatus(200)
            .matchBodySnapshot()
            .matchHeaderSnapshot({
                etag: anyEtag
            });

        mockManager.assert.sentEmail({
            to: 'joe@example.com',
            subject: 'testemail'
        });
    });

    it('Can send a test mail', async function () {
        // @TODO: either remove this endpoint or fix its response body
        await agent
            .post('mail/test')
            .expectStatus(200)
            .matchBodySnapshot()
            .matchHeaderSnapshot({
                etag: anyEtag
            });

        mockManager.assert.sentEmail({
            to: 'jbloggs@example.com',
            subject: 'Test Ghost Email'
        });
    });
});
