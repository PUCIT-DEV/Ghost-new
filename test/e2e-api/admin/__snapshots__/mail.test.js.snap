// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mail API Can send a test mail 1: [body] 1`] = `
Object {
  "message": "sent",
}
`;

exports[`Mail API Can send a test mail 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": "18",
  "content-type": "application/json; charset=utf-8",
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Origin, Accept-Encoding",
  "x-powered-by": "Express",
}
`;

exports[`Mail API Can send mail 1: [body] 1`] = `
Object {
  "mail": Array [
    Object {
      "message": Object {
        "html": "<p>This</p>",
        "subject": "testemail",
        "to": "joe@example.com",
      },
      "status": Object {
        "message": "sent",
      },
    },
  ],
}
`;

exports[`Mail API Can send mail 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": "118",
  "content-type": "application/json; charset=utf-8",
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Origin, Accept-Encoding",
  "x-powered-by": "Express",
}
`;
